src/partition/ParquetInterface.cc

```diff
@@ -0,0 +1,353 @@
+}
+
+// Read a batch from the file and format the table iaccording to the partition configuration file
+//    -> column reordering, true/false -> 1/0, remove null values, ...
+arrow::Status ParquetFile::ReadNextBatchTable_Formatted(std::shared_ptr<arrow::Table>& outputTable) {
+    arrow::Result<std::shared_ptr<arrow::RecordBatch>> maybe_batch = m_rb_reader_gbl->Next();
+
+    std::vector<std::string> paramNotFound;
+    std::map<std::string, std::shared_ptr<arrow::Field>> fieldConfig;
+
+    if (maybe_batch != nullptr) {
+        // DumpProcessMemory("VmRSS", true);
+
+        ARROW_ASSIGN_OR_RAISE(auto batch, maybe_batch);
+        std::shared_ptr<arrow::Table> initTable;
+        ARROW_ASSIGN_OR_RAISE(initTable, arrow::Table::FromRecordBatches(batch->schema(), {batch}));
+
+        // Increment the batch number
+        m_batchNumber++;
+
+        const arrow::FieldVector fields = initTable->schema()->fields();
+        for (auto fd : fields) {
+            fieldConfig[fd->name()] = fd;
+        }
+
+        arrow::FieldVector formatedTable_fields;
+        std::vector<std::shared_ptr<arrow::ChunkedArray>> formatedTable_columns;
+
+        // Loop over the column names as defined in the partition config file
+        for (std::string paramName : m_parameterNames) {
+            std::shared_ptr<arrow::ChunkedArray> chunkedArray = initTable->GetColumnByName(paramName);
+
+            // Column not found in the arrow table...
+            if (chunkedArray == NULLPTR) {
+                paramNotFound.push_back(paramName);
+            } else {
+                // Column type is boolean -> switch to 0/1 representation
+                if (fieldConfig[paramName]->type() == arrow::boolean()) {
+                    auto newChunkedArray = ChunkArrayReformatBoolean(chunkedArray, true);
+                    if (newChunkedArray == nullptr) {
+                        return arrow::Status::ExecutionError("Error while formating boolean chunk array");
+                    }
+                    formatedTable_columns.push_back(newChunkedArray);
+
+                    std::shared_ptr<arrow::Field> newField =
+                            std::make_shared<arrow::Field>(fieldConfig[paramName]->name(), arrow::int8());
+                    formatedTable_fields.push_back(newField);
+                }
+                // Simply keep the chunk as it is defined in teh arrow table
+                else {
+                    formatedTable_columns.push_back(chunkedArray);
+                    formatedTable_fields.push_back(fieldConfig[paramName]);
+                }
+            }
+        }  // end of loop over parameters
+
+        // If a column is not found (i.e. a parameter defined in partition.json does not exist in parquet
+        // file), throw an error and stop
+        if (paramNotFound.size() > 0) {
+            for (auto name : paramNotFound)
+                std::cout << "ERROR : param name " << name << " not found in table columns" << std::endl;
```

Comment:
`std::cout` is not allowed in Qserv code. Please, switch to using LSST Logging service.

Reply:
Done